src/sentry/tasks/integrations.py

```diff
@@ -203,3 +207,63 @@ def migrate_repo(repo_id, integration_id, organization_id):
             integration=integration,
             organization=Organization.objects.get(id=organization_id),
         )
+
+
+@instrumented_task(
+    name='sentry.tasks.integrations.kickoff_vsts_subscription_check',
+    queue='integrations',
+    default_retry_delay=60 * 5,
+    max_retries=5,
+)
+@retry()
+def kickoff_vsts_subscription_check():
+    organization_integrations = OrganizationIntegration.objects.filter(
+        integration__provider='vsts',
+        integration__status=ObjectStatus.VISIBLE,
+        status=ObjectStatus.VISIBLE,
+    ).select_related('integration')
+    six_hours_ago = time() - timedelta(hours=6).seconds
+    for org_integration in organization_integrations:
+        organization_id = org_integration.organization_id
+        integration = org_integration.integration
+
+        try:
+            if 'subscription' not in integration.metadata or integration.metadata[
+                    'subscription']['check'] > six_hours_ago:
+                continue
+        except KeyError:
+            pass
+
+        vsts_subscription_check.apply_async(
+            kwargs={
+                'integration_id': integration.id,
+                'organization_id': organization_id,
+            }
+        )
+
+
+@instrumented_task(
+    name='sentry.tasks.integrations.vsts_subscription_check',
+    queue='integrations',
+    default_retry_delay=60 * 5,
+    max_retries=5,
+)
+@retry(exclude=(ApiError, ApiUnauthorized, Integration.DoesNotExist))
+def vsts_subscription_check(integration_id, organization_id, **kwargs):
+    integration = Integration.objects.get(id=integration_id)
+    installation = integration.get_installation(organization_id=organization_id)
```

Comment:
Do you have to worry about missing records here?

Reply:
@markstory I believe if the query fails, it'll simply not retry the `vsts_subscription_check` again for that particular integration (but would six hours later....). But I should probably be logging that somewhere. Does that actually show up in sentry I wonder? I assume so?

Reply:
@markstory Now that I think on it, I'm not sure logging would do that much good here. If the integration doesn't exist it's not really actionable here.